### PR TITLE
added http-x-forwarded-for field to nginx parser

### PR DESCRIPTION
**Pull request by yaron-idan**
Closing #1907, an update to the docs has been proposed in https://github.com/fluent/fluentd-docs/issues/471.

### CODE REVIEW

File: test/plugin/test_parser_nginx.rb [hunk 1]
```diff
@@ -24,23 +25,23 @@ def create_driver
 
   def test_parse
     d = create_driver
-    d.instance.parse('127.0.0.1 192.168.0.1 - [28/Feb/2013:12:00:00 +0900] "GET / HTTP/1.1" 200 777 "-" "Opera/12.0"') { |time, record|
+    d.instance.parse('127.0.0.1 192.168.0.1 - [28/Feb/2013:12:00:00 +0900] "GET / HTTP/1.1" 200 777 "-" "Opera/12.0" -') { |time, record|
```
Comment (repeatedly) [inline test/plugin/test_parser_nginx.rb]: Don't change existing test because it is needed to check compatibility.

Comment (yaron-idan) [reply test/plugin/test_parser_nginx.rb]: I see, this test will fail though if it won't be adapted to the nginx log format, how are you suggesting to overcome this issue? Create another test with the new format?

Comment (repeatedly) [reply test/plugin/test_parser_nginx.rb]: > Create another test

Yes, creating new test is needed. Many users use nginx format so we can't break it. It breaks user environment and causes log lost.

> the new format?

One regular expression seems to support log with/without http_x_forwarded_for.

**Comment by yaron-idan** [comment]
I've fixed the REGEXP to make the new field optional. I've also heed the advice of an SO answer and swapped `[^ ]` with `\S` which seems more readable to me - but I can change back if you don't agree.
All existing tests pass fine, and a new test has been added to check compatibility with the extended log entry. Hope that fits the requested changes.

**Comment by repeatedly** [comment]
`[^ ]` and `\S` are different and `\S` is bit slower. Please back changes to previous implementation.
In addition, we enabled DCO for our repository. Please add your sign to the commit. See https://github.com/apps/dco

**Comment by yaron-idan** [comment]
OK, seems I'm gonna have to open a new PR since DCO rejects commits without a sign off, I'm gonna close this one and open a new one with the changes requested + a sign off.

**Pull request closed without merging**